# Importing the libraries

In [22]:
import pandas as pd
import numpy as np
import plotly.express as px

In [3]:
df = pd.read_csv("proteinGroups.csv", low_memory=False)
df.head(10)

Protein IDs Majority protein IDs Peptide counts (all)  \
0  A0A024RBG1           A0A024RBG1                    7   
1  A0A096LP55               P07919           A0A096LP55   
2      P0DPI2           A0A0B4J2D5               P0DPI2   
3      P0DPI3           A0A0U1RR11           A0A0U1RRI6   
4      A0AV96               Q8TBY0               A0AV96   
5      A0AVF1               A0AVF1                    5   
6      A0AVT1               A0AVT1                   35   
7      A0FGR8               A0FGR8                   24   
8      A0JLT2               A0JLT2                    3   
9      A0JNW5               A0JNW5                    3   

  Peptide counts (razor+unique) Peptide counts (unique)  \
0                             2                       2   
1                        P07919                       1   
2                    A0A0B4J2D5                       7   
3                        P0DPI3              A0A0U1RR11   
4                            13                       1   
5                             5                       5   
6                            35                      35   
7                            24                      24   
8                             3                       3   
9                             3                       3   

                                       Protein names Gene names Fasta headers  \
0                                                NaN      NUDT4           NaN   
1                                                  1          1             1   
2                                                  7          7             7   
3                                         A0A0U1RRI6          1             1   
4                                                 13          1            12   
5                Intraflagellar transport protein 56      TTC26           NaN   
6        Ubiquitin-like modifier-activating enzyme 6       UBA6           NaN   
7                           Extended synaptotagmin-2      ESYT2           NaN   
8  Mediator of RNA polymerase II transcription su...      MED19           NaN   
9                       UHRF1-binding protein 1-like  UHRF1BP1L           NaN   

  Number of proteins                Peptides  ... Unnamed: 1014 Unnamed: 1015  \
0                  1                       7  ...           NaN           NaN   
1                  1                       1  ...           NaN           NaN   
2                  7                       7  ...           NaN           NaN   
3                  1                       1  ...           NaN           NaN   
4                  0  RNA-binding protein 47  ...           NaN           NaN   
5                  1                       5  ...           NaN           NaN   
6                  1                      35  ...           NaN           NaN   
7                  1                      24  ...           NaN           NaN   
8                  1                       3  ...           NaN           NaN   
9                  1                       3  ...           NaN           NaN   

  Unnamed: 1016 Unnamed: 1017 Unnamed: 1018 Unnamed: 1019 Unnamed: 1020  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   
2           NaN           NaN           NaN           NaN           NaN   
3           NaN           NaN           NaN           NaN           NaN   
4           NaN           NaN           NaN           NaN           NaN   
5           NaN           NaN           NaN           NaN           NaN   
6           NaN           NaN           NaN           NaN           NaN   
7           NaN           NaN           NaN           NaN           NaN   
8           NaN           NaN           NaN           NaN           NaN   
9           NaN           NaN           NaN           NaN           NaN   

  Unnamed: 1021 Unnamed: 1022 Unnamed: 1023  
0           NaN           NaN           NaN  


# How many proteins were detected in total?

In [4]:
df['Protein IDs'].nunique()

7499

In [6]:
df2 = df[~df['Protein IDs'].str.startswith('CON')]
df3 = df2[~df2['Protein IDs'].str.startswith('REV')]
df3['Protein IDs'].nunique()

7251

In [8]:
df3['Protein IDs'].value_counts().sum()

7251

In [9]:
df3.shape

(7251, 1024)

In [16]:
df3.to_csv("my-sample.csv",index_label=False)

In [10]:
df4 = df3['Protein IDs'].dropna(axis=0)

In [11]:
df4.shape

(7251,)

So, total number of unique protein presents are 7251

In [25]:
df3.head(10)

Protein IDs Majority protein IDs Peptide counts (all)  \
0  A0A024RBG1           A0A024RBG1                    7   
1  A0A096LP55               P07919           A0A096LP55   
2      P0DPI2           A0A0B4J2D5               P0DPI2   
3      P0DPI3           A0A0U1RR11           A0A0U1RRI6   
4      A0AV96               Q8TBY0               A0AV96   
5      A0AVF1               A0AVF1                    5   
6      A0AVT1               A0AVT1                   35   
7      A0FGR8               A0FGR8                   24   
8      A0JLT2               A0JLT2                    3   
9      A0JNW5               A0JNW5                    3   

  Peptide counts (razor+unique) Peptide counts (unique)  \
0                             2                       2   
1                        P07919                       1   
2                    A0A0B4J2D5                       7   
3                        P0DPI3              A0A0U1RR11   
4                            13                       1   
5                             5                       5   
6                            35                      35   
7                            24                      24   
8                             3                       3   
9                             3                       3   

                                       Protein names Gene names Fasta headers  \
0                                                NaN      NUDT4           NaN   
1                                                  1          1             1   
2                                                  7          7             7   
3                                         A0A0U1RRI6          1             1   
4                                                 13          1            12   
5                Intraflagellar transport protein 56      TTC26           NaN   
6        Ubiquitin-like modifier-activating enzyme 6       UBA6           NaN   
7                           Extended synaptotagmin-2      ESYT2           NaN   
8  Mediator of RNA polymerase II transcription su...      MED19           NaN   
9                       UHRF1-binding protein 1-like  UHRF1BP1L           NaN   

  Number of proteins                Peptides  ... Unnamed: 1014 Unnamed: 1015  \
0                  1                       7  ...           NaN           NaN   
1                  1                       1  ...           NaN           NaN   
2                  7                       7  ...           NaN           NaN   
3                  1                       1  ...           NaN           NaN   
4                  0  RNA-binding protein 47  ...           NaN           NaN   
5                  1                       5  ...           NaN           NaN   
6                  1                      35  ...           NaN           NaN   
7                  1                      24  ...           NaN           NaN   
8                  1                       3  ...           NaN           NaN   
9                  1                       3  ...           NaN           NaN   

  Unnamed: 1016 Unnamed: 1017 Unnamed: 1018 Unnamed: 1019 Unnamed: 1020  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   
2           NaN           NaN           NaN           NaN           NaN   
3           NaN           NaN           NaN           NaN           NaN   
4           NaN           NaN           NaN           NaN           NaN   
5           NaN           NaN           NaN           NaN           NaN   
6           NaN           NaN           NaN           NaN           NaN   
7           NaN           NaN           NaN           NaN           NaN   
8           NaN           NaN           NaN           NaN           NaN   
9           NaN           NaN           NaN           NaN           NaN   

  Unnamed: 1021 Unnamed: 1022 Unnamed: 1023  
0           NaN           NaN           NaN  


In [41]:
df5 = df3.iloc[:,24:30]

In [42]:
df5

Unique peptides SW480_1 Unique peptides SW480_2 Unique peptides SW480_3  \
0                          2                       2                       2   
1                          0                       1                       1   
2                          7                       6                       7   
3                          1                       1                       1   
4                          4                      12                      13   
...                      ...                     ...                     ...   
7331                       4                       1                       3   
7332                       5                       5                       5   
7333                       6                       6                       7   
7334                       2                       4                       3   
7335                      17                      22                      21   

     Unique peptides SW620_1 Unique peptides SW620_2 Unique peptides Sw620_3  
0                          2                       2                       2  
1                          1                       1                       1  
2                          7                       7                       7  
3                          1                       1                       1  
4                         12                       1                       3  
...                      ...                     ...                     ...  
7331                       5                       4                       6  
7332                       1                       2                       2  
7333                       8                       5                       6  
7334                       3                       3                       3  
7335                      21                      21                      23  

[7251 rows x 6 columns]

In [45]:
df6.head(20)

Unique peptides SW480_1 Unique peptides SW480_2 Unique peptides SW480_3  \
0                        2                       2                       2   
1                        0                       1                       1   
2                        7                       6                       7   
3                        1                       1                       1   
4                        4                      12                      13   
5                        4                       4                       4   
6                       32                      33                      33   
7                       24                      23                      24   
8                        3                       3                       3   
9                        3                       3                       3   
10                      16                      20                      19   
11                       1                       1                       1   
12                       0                       0                       0   
13                       9                      11                      11   
14                       4                       3                       4   
15                       3                       3                       2   
16                       1                       1                       1   
17                       0                       0                       0   
18                  B4DXR9                  Q03936                  Q96N58   
19                      24                      25                      25   

   Unique peptides SW620_1 Unique peptides SW620_2 Unique peptides Sw620_3  
0                        2                       2                       2  
1                        1                       1                       1  
2                        7                       7                       7  
3                        1                       1                       1  
4                       12                       1                       3  
5                        5                       4                       4  
6                       33                      32                      33  
7                       23                      23                      22  
8                        3                       3                       3  
9                        2                       3                       2  
10                      23                      27                      25  
11                       1                       0                       1  
12                       0                       1                       0  
13                      11                      11                      10  
14                       4                       4                       4  
15                       3                       4                       4  
16                       1                       1                       1  
17                       0                       0                       0  
18                  Q8N4W9                  Q05481                  A6NN14  
19                      23                      22                      26

In [58]:
df5.shape

(7251, 6)

In [66]:
df5 = df5[np.isfinite(pd.to_numeric(df5['Unique peptides SW480_1'], errors="coerce"))]
df5 = df5[np.isfinite(pd.to_numeric(df5['Unique peptides SW480_2'], errors="coerce"))]
df5 = df5[np.isfinite(pd.to_numeric(df5['Unique peptides SW480_3'], errors="coerce"))]
df5 = df5[np.isfinite(pd.to_numeric(df5['Unique peptides SW620_1'], errors="coerce"))]
df5 = df5[np.isfinite(pd.to_numeric(df5['Unique peptides SW620_2'], errors="coerce"))]
df5 = df5[np.isfinite(pd.to_numeric(df5['Unique peptides Sw620_3'], errors="coerce"))]
df5.head(10)

Unique peptides SW480_1 Unique peptides SW480_2 Unique peptides SW480_3  \
0                       2                       2                       2   
1                       0                       1                       1   
2                       7                       6                       7   
3                       1                       1                       1   
4                       4                      12                      13   
5                       4                       4                       4   
6                      32                      33                      33   
7                      24                      23                      24   
8                       3                       3                       3   
9                       3                       3                       3   

  Unique peptides SW620_1 Unique peptides SW620_2 Unique peptides Sw620_3  
0                       2                       2                       2  
1                       1                       1                       1  
2                       7                       7                       7  
3                       1                       1                       1  
4                      12                       1                       3  
5                       5                       4                       4  
6                      33                      32                      33  
7                      23                      23                      22  
8                       3                       3                       3  
9                       2                       3                       2

In [84]:
df5['Unique peptides SW480_1'] = df5['Unique peptides SW480_1'].astype('int')
df5['Unique peptides SW480_2'] = df5['Unique peptides SW480_2'].astype('int')
df5['Unique peptides SW480_3'] = df5['Unique peptides SW480_3'].astype('int')
df5['Unique peptides SW620_1'] = df5['Unique peptides SW620_1'].astype('int')
df5['Unique peptides SW620_2'] = df5['Unique peptides SW620_2'].astype('int')
df5['Unique peptides Sw620_3'] = df5['Unique peptides Sw620_3'].astype('int')

In [85]:
df5.head(20)

Unique peptides SW480_1  Unique peptides SW480_2  Unique peptides SW480_3  \
0                         2                        2                        2   
1                         0                        1                        1   
2                         7                        6                        7   
3                         1                        1                        1   
4                         4                       12                       13   
5                         4                        4                        4   
6                        32                       33                       33   
7                        24                       23                       24   
8                         3                        3                        3   
9                         3                        3                        3   
10                       16                       20                       19   
11                        1                        1                        1   
12                        0                        0                        0   
13                        9                       11                       11   
14                        4                        3                        4   
15                        3                        3                        2   
16                        1                        1                        1   
17                        0                        0                        0   
19                       24                       25                       25   
20                        3                        3                        3   

    Unique peptides SW620_1  Unique peptides SW620_2  Unique peptides Sw620_3  
0                         2                        2                        2  
1                         1                        1                        1  
2                         7                        7                        7  
3                         1                        1                        1  
4                        12                        1                        3  
5                         5                        4                        4  
6                        33                       32                       33  
7                        23                       23                       22  
8                         3                        3                        3  
9                         2                        3                        2  
10                       23                       27                       25  
11                        1                        0                        1  
12                        0                        1                        0  
13                       11                       11                       10  
14                        4                        4                        4  
15                        3                        4                        4  
16                        1                        1                        1  
17                        0                        0                        0  
19                       23                       22                       26  
20                        3                        2                        3

In [83]:
df5['Unique peptides SW480_1'].as

dtype('O')

In [86]:
Sum = df5.sum(axis=0)
Sum

Unique peptides SW480_1    66490
Unique peptides SW480_2    66743
Unique peptides SW480_3    67976
Unique peptides SW620_1    63554
Unique peptides SW620_2    65655
Unique peptides Sw620_3    65463
dtype: int64

In [92]:
df_sum = Sum.to_frame().reset_index()
df_sum.columns = ['Cell_Type','Total_Sum']
df_sum

Cell_Type  Total_Sum
0  Unique peptides SW480_1      66490
1  Unique peptides SW480_2      66743
2  Unique peptides SW480_3      67976
3  Unique peptides SW620_1      63554
4  Unique peptides SW620_2      65655
5  Unique peptides Sw620_3      65463

In [94]:
fig = px.bar(df_sum,x='Cell_Type',y='Total_Sum',title="Bar Chart for SW480 & SW620 Cell", barmode='group')
fig.show()